zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -26,14 +34,81 @@ pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNeg
     }
 }
 
+/// Get the founders reward t-address for the specified block height as described in [protocol specification §7.8][7.8]
+///
+/// [7.8]: https://zips.z.cash/protocol/canopy.pdf#foundersreward
+pub fn founders_reward_address(height: Height, network: Network) -> Result<String, Error> {
+    let blossom_height = Blossom
+        .activation_height(network)
+        .expect("blossom activation height should be available");
+    let canopy_height = Canopy
+        .activation_height(network)
+        .expect("canopy activation height should be available");
+
+    if height >= canopy_height {
+        panic!("no address returned after canopy");
+    }
+
+    let mut adjusted_height = height;
+    if height >= blossom_height {
+        adjusted_height = Height(
+            blossom_height.0
+                + ((height.0 - blossom_height.0) / (BLOSSOM_POW_TARGET_SPACING_RATIO as u32)),
+        );
+    }
+
+    let address_index = 1 + (adjusted_height.0 / FOUNDER_ADDRESS_CHANGE_INTERVAL as u32);
+
+    let mut addresses = FOUNDERS_REWARD_ADDRESSES_MAINNET;
+    if network == Network::Testnet {
+        addresses = FOUNDERS_REWARD_ADDRESSES_TESTNET;
+    }
+    Ok(addresses[(address_index - 1) as usize].to_string())
+}
+
+/// Returns a list of outputs in `Transaction`, which have a script address equal to `String`.
+pub fn find_output_with_address(transaction: &Transaction, address: &str) -> Vec<Output> {
+    let calculated_addr: Address = address.parse().unwrap();
+
+    // For debugging
+    println!("Looking for founders reward address in new coinbase transaction:");
+    for (output_number, o) in transaction.outputs().iter().enumerate() {
+        let lock_script = o.lock_script.clone();
+
+        // For one of the coinbase outputs the address we get from the lock_script should be one of
+        // the ones in the FOUNDERS_REWARD_ADDRESSES_MAINNET array, not happening.
+        let output_address: Address = Address::from(lock_script);
+
+        println!(
+            "Output: {} Calculated address: {} Output address: {}",
+            output_number, calculated_addr, output_address
+        );
+        println!();
```

Comment:
```suggestion
```suggestion
    tracing::debug!(?output_number, ?calculated_addr, ?output_address);
```
```